### ISSUE TITLE
Bug: [@typescript-eslint/no-unused-vars] vscode report unused, but run eslint not report

### ISSUE THREAD

**Issue by takeseem** [typescript-eslint/typescript-eslint#11540]
it's only report 'abc' unused, but vscode can report correct all var unused.

`44:17  error  'abc' is defined but never used  @typescript-eslint/no-unused-vars`

```ts
private havFun(abc?: string) {
	this.winHandlers.forEach((handler, win) => {
		if (win.closed) {
			this.logger.debug('check: win is closed, call unbind');
			this.unbind(win);
		}
	});
}
```

may be `win` is used, so it's ignore `handler` check.
I change code to use `handler`, now it can report `win` not used.

```ts
private havFun(abc?: string) {
	this.winHandlers.forEach((handler, win) => {
		if (typeof handler !== 'function') {
			this.logger.debug('check: win is closed, call unbind');
			// this.unbind(win);
		}
	});
}
```

all config is default

**Comment by JoshuaKGoldberg** [comment]
> all config is default

There is no "default" config 🙂 ESLint is very configurable. Plus you've mentioned an editor, VS Code, being used. We can't possibly help you if you don't explain how to reproduce the behavior you're seeing.

https://antfu.me/posts/why-reproductions-are-required

Help us help you, please!

**Comment by takeseem** [comment]
my problem, i create a project for this [eslint-test](https://github.com/takeseem/eslint-test/blob/d37e048ca23793ff23d5f2ff61a481cac510949d/src/lib/test.ts#L18)

reproduce: check out and run `pnpm lint`, eslint can not report `handler` unused in `havFun2` function.

<img width="960" height="1062" alt="reproduce" src="https://github.com/user-attachments/assets/4a5c9f5c-280d-4489-b12d-5f2793d96b3e" />

**Comment by takeseem** [comment]
I add `havFun3`: 	https://github.com/takeseem/eslint-test/blob/8bea3afd6f61414d3e74b91f0094f96a5c74b7de/src/lib/test.ts#L26
```
	/** command eslint cannot report any unused var */
	havFun3(a: string, b: string, c: string, last: string) {
		if (last != '1') {
			console.log('check: last is not 1');
		}
	}
```

**Comment by takeseem** [comment]
I add [havFun3](https://github.com/takeseem/eslint-test/blob/8bea3afd6f61414d3e74b91f0094f96a5c74b7de/src/lib/test.ts#L26), verify that:
if the last var is used, the any before args cannot to check unused.
```
	/** command eslint cannot report any unused var */
	havFun3(a: string, b: string, c: string, last: string) {
		if (last != '1') {
			console.log('check: last is not 1');
		}
	}
```

**Comment by bradzacher** [comment]
https://eslint.org/docs/latest/rules/no-unused-vars#args

Please read the docs this is working as expected.

**Comment by takeseem** [comment]
Thank you.  I'm sorry that.  please close the issuse.
